src/dmd/dtoh.d

```diff
@@ -180,6 +180,21 @@ struct _d_dynamicArray final
 `);
     }
 
+    // prevent trailing newlines
+    foreach_reverse (i, ch; buf)
+    {
+        if (ch != '\n')
+        {
+            if (buf.length)
+            {
+                size_t count = (buf.length - 1) - i;
+                if (count)
+                    buf.remove(buf.length - 1, count - 1);
```

Comment:
Iteration + modification will work with the current `Outbuffer` implementation but it's probably cleaner to determine the proper end index + resize at the end.

Reply:
This will not iterate, when `if (ch != '\n')` is triggered, the code will only run once.